reify.go

```diff
@@ -85,8 +85,12 @@ import (
 //  The struct tag options `replace`, `append`, and `prepend` overwrites the
 //  global value merging strategy (e.g. ReplaceValues, AppendValues, ...) for all sub-fields.
 //
+// When unpacking into a map, primitive, or struct Unpack will call InitDefaults if
+// the type implements the Initializer interface. The Initializer interface is not supported
+// on arrays or slices.
+//
 // Fields available in a struct or a map, but not in the Config object, will not
-// be touched. Default values should be set in the target value before calling Unpack.
+// be touched by Unpack unless they are initialized from InitDefaults.
```

Comment:
Let's add that InitDefault is executed top-down, not bottom-up.

e.g. if we have `type A struct { B B}`, with both A, and B implementing InitDefaults, then `A.InitDefaults` is called before `B.InitDefaults`. This means that `A.InitDefaults` can not overwrite defaults set via `B.InitDefaults`.